hq/app/logic/SnykDisplay.scala

```diff
@@ -1,81 +1,74 @@
 package logic
 
-import com.fasterxml.jackson.core.JsonParseException
 import play.api.libs.json._
-import utils.attempt.{Attempt, FailedAttempt, Failure}
+import utils.attempt.{Attempt, Failure}
 
 import scala.concurrent.ExecutionContext
 import model._
 import model.Serializers._
 
+import scala.util.{Success, Try}
+
 object SnykDisplay {
 
-  def getOrganisation(s: String, requiredOrganisation: SnykOrganisationName): Attempt[SnykOrganisation] = {
-    val organisation = for {
-      orglist <- parseJsonToOrganisationList(s)
-      org <- orglist.find(_.name == requiredOrganisation.value)
-    } yield org
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find organisation from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(organisation, FailedAttempt(f))
+  def parseOrganisations(rawJson: String, snykGroupId: SnykGroupId)(implicit ec: ExecutionContext): Attempt[List[SnykOrganisation]] = {
+    for {
+      organisationList <- parseJsonToOrganisationList(rawJson)
+      guardianOrganisationList = organisationList filter {
+        case SnykOrganisation(_, _, Some(group)) => group.id==snykGroupId.value
+        case _ => false
+      }
+    } yield guardianOrganisationList
   }
 
-  private def parseJsonToError(s: String): SnykError = try {
-    Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
-  } catch {
-    case _: JsonParseException => SnykError(s)
+  private def parseJsonToError(s: String): SnykError = Try(Json.parse(s).validateOpt[SnykError]) match {
+    case Success(JsSuccess(Some(error), _)) => error
+    case _ => SnykError(s)
   }
 
-  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
-    (Json.parse(s) \ "orgs").asOpt[List[SnykOrganisation]]
-  } catch {
-    case _: JsonParseException => None
+  def parseJsonToObject[A](label: String, rawJson: String, f: String => JsResult[A]): Attempt[A] = Try(f(rawJson)) match {
+    case Success(JsSuccess(parsedObject, _)) => Attempt.Right(parsedObject)
+    case Success(JsError(e)) =>
+      val error = parseJsonToError(rawJson)
+      val failure = Failure(s"Unable to find $label from $rawJson: $e", s"Could not read Snyk response (${error.error})", 502, None, None)
+      Attempt.Left(failure)
+    case scala.util.Failure(e) =>
```

Comment:
~Better to import Failure, no need to full qualify it in the case statement.~

Ah, I see it is because it clashes name.